### PR TITLE
Add example for MSP432E401Y Launchpad for Ethernet

### PR DESCRIPTION
**Pull request by 9names**
MSP432E401Y is (as far as I know) just TM4C1294NCPD with different functions in ROM.
Since this HAL does not use the SimpleLink ROM functions, this chip should be 100% compatible with [tm4c129x-hal](https://github.com/rust-embedded-community/tm4c-hal/tree/master/tm4c129x-hal)

I've tested it out UART and GPIO on the launchpad. Everything is working fine so far.
Wasn't sure if you'd accept something like this in your HAL - I'm happy to keep this example project as a separate thing on my github account, but I figured it was worth making a PR just in case.

Note that the default runner is `probe-rs run`, which does not support this chip until https://github.com/probe-rs/probe-rs/pull/1139 is merged.

### CODE REVIEW

File: examples/msp432-launchpad-for-ethernet/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[package]
+authors = [
+	"9names",
+]
+edition = "2018"
+readme = "README.md"
+name = "msp432-launchpad-for-ethernet"
+version = "0.1.0"
```
Comment (thejpster) [inline examples/msp432-launchpad-for-ethernet/Cargo.toml:8]: Is there a licence we can add here?

File: examples/msp432-launchpad-for-ethernet/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# MSP432 Launchpad for Ethernet Demo Application
+
+> For the MSP432 Launchpad developement kit for Ethernet SimpleLink MCU, [MSP-EXP432E401Y](https://www.ti.com/tool/MSP-EXP432E401Y)
+
+## License
+
+This template is licensed under either of
```
Comment (thejpster) [inline examples/msp432-launchpad-for-ethernet/README.md:7]: s/template/example

**Review by thejpster — CHANGES_REQUESTED** [review]
Couple of tiny nits but otherwise good to go!

File: examples/msp432-launchpad-for-ethernet/Cargo.toml (hunk 1 above)
Comment (9names) [reply examples/msp432-launchpad-for-ethernet/Cargo.toml:8]: Added licenses to match readme and source material

File: examples/msp432-launchpad-for-ethernet/README.md (hunk 2 above)
Comment (9names) [reply examples/msp432-launchpad-for-ethernet/README.md]: Fixed, thanks!

**Review by thejpster — APPROVED** [review]

**Pull request merged**